src/planscape/stands/management/commands/calculate_metrics.py

```diff
@@ -0,0 +1,205 @@
+import json
+import multiprocessing
+import time
+from itertools import repeat
+
+import rasterio
+from conditions.models import Condition
+from conditions.registry import get_raster_path
+from django.contrib.gis.db.models.functions import AsWKT, Transform
+from django.contrib.gis.geos import GEOSGeometry
+from django.core.management.base import BaseCommand
+from django.db import connection
+from stands.stats import get_zonal_stats
+from stands.models import Stand
+from django.conf import settings
+
+
+def calculate_metric(stand, condition_id, raster):
+    stand_id, geometry = stand
+    stats = get_zonal_stats(
+        stand_geometry=geometry,
+        raster=raster,
+    )
+
+    if stats["count"] <= 0:
+        return None
+
+    min = stats["min"]
+    max = stats["max"]
+    avg = stats["mean"]
+    sum = stats["sum"]
+    maj = stats["majority"]
+    mnt = stats["minority"]
+    count = stats["count"]
+
+    return f"{stand_id},{condition_id},{min},{max},{avg},{sum},{count},{maj},{mnt}\n"
+
+
+class Command(BaseCommand):
+    help = "Calculates stand metrics based on conditions existing in the database and creates a file to be copied."
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--condition-ids",
+            nargs="+",
+            type=int,
+        )
+
+        parser.add_argument(
+            "--size",
+            type=str,
+            default="LARGE",
+        )
+
+        parser.add_argument(
+            "--outfile",
+            type=str,
+        )
+
+        parser.add_argument("--discover", type=str, default=None)
+
+        parser.add_argument("--max-workers", type=int, default=4)
+
+        parser.add_argument("--force-yes", action="store_true")
+
+    def get_condition_extent(self, condition_id):
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT
+                    ST_Envelope(ST_Collect(ST_Envelope(raster))) AS "geometry"
+                FROM conditions_conditionraster cc 
+                WHERE condition_id = %s""",
+                [condition_id],
+            )
+
+            row = cursor.fetchone()
+            geometry = row[0]
+            if geometry:
+                return GEOSGeometry(row[0])
+            return None
+
+    def get_stands_queryset(self, condition_id=None, size=None):
+        queryset = Stand.objects.all()
+
+        if size:
+            queryset = queryset.filter(size=size)
+
+        if condition_id:
+            extent = self.get_condition_extent(condition_id)
+            if not extent:
+                self.stdout.write(
+                    f"[WARN] Empty extent for condition {condition_id}. No metrics to load."
+                )
+                return Stand.objects.none()
+
+            queryset = queryset.filter(geometry__intersects=extent)
+
+        return queryset.annotate(
+            geom=AsWKT(Transform(srid=3857, expression="geometry"))
```

Comment:
Any chance we'll change this SRID? Would there be any benefit to defining it as a constant, in case we ever want to?